ostd/src/util/vaddr_alloc.rs

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::collections::btree_map::BTreeMap;
+use core::ops::Range;
+
+use crate::{
+    prelude::*,
+    sync::{PreemptDisabled, SpinLock, SpinLockGuard},
+    Error,
+};
+
+pub struct KVirtAreaFreeNode {
```

Comment:
This is mostly an internal type that is not used anywhere else than this file. So the `pub` should be removed.